### PR TITLE
kubevirt: Add edit VM flavor integration test

### PR DESCRIPTION
**Pull request by mareklibra**
Follow-up for #2392.

Depends on:
- [x] #2812 

Please note, just last commit is relevant.

[Screen-cast of the integration test execution]( https://youtu.be/DsfgPrwg-ZA)

### CODE REVIEW

**Comment by mareklibra** [comment]
Rebased as master-4.3 is created now.
Depends on #2392.

**Comment by mareklibra** [comment]
@rhrazdil , can you please have a look? Just last 3 commits are relevant.

When rebasing, I noticed a bunch of changes in the integration tests since I wrote this PR originally and waited for the master-4.3 to be created. Aren't my changes in conflict with the intention of other PRs, please?

**Comment by mareklibra** [comment]
Rebased as https://github.com/openshift/console/pull/2392 is merged now.

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.flavor.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+          await virtualMachineView.vmDetailLabelValue('flavor.template.kubevirt.io/large'),
+        ).toBe('true');
+        expect(
+          (await virtualMachineView.vmDetailLabelValue('vm.kubevirt.io/template')).startsWith(
+            'rhel7-desktop-tiny-', // template is not changed (might be in the future)
+          ),
+        ).toBeTruthy();
+      });
+    },
+    VM_BOOTUP_TIMEOUT_SECS,
```
Comment (rhrazdil) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.flavor.ts]: Maybe we should come up with some more specific timeout for this.

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.wizard.configs.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    attached: true,
+  };
+};
+
+export const CONFIG_NAME_URL = 'URL';
+export const CONFIG_NAME_CONTAINER = 'Container';
+export const CONFIG_NAME_PXE = 'PXE';
+export const CONFIG_NAME_CLONED_DISK = 'ClonedDisk';
+
+export const getProvisionConfigs = (testName: string) =>
```
Comment (rhrazdil) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/vm.wizard.configs.ts]: +1, nice

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.flavor.ts (hunk 1 above)
Comment (mareklibra) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.flavor.ts]: done: `VM_CREATE_AND_EDIT_TIMEOUT_SECS`, wdyt?

Comment (rhrazdil) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.flavor.ts]: Looks good.

**Comment by mareklibra** [comment]
I have splited this PR to simplify review.

**Comment by mareklibra** [comment]
rebased as #2812 is merged now

**Comment by mareklibra** [comment]
/hold 
for #2817

**Comment by mareklibra** [comment]
retest

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.flavor.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+          until.textToBePresentInElement(
+            virtualMachineView.vmDetailFlavor(vm.namespace, vm.name),
+            'Tiny: 1 vCPU, 1 GB Memory',
+          ),
+        );
+        await vm.modalEditFlavor();
+        await browser.wait(
+          until.textToBePresentInElement(editFlavorView.flavorDropdownText, 'Tiny'),
+        );
+        await selectDropdownOptionById('#vm-flavor-modal-flavor-dropdown', 'large-link');
```
Comment (rhrazdil) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.flavor.ts]: Can we define this in virtualmachine.view.ts file?

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.flavor.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            virtualMachineView.vmDetailFlavor(vm.namespace, vm.name),
+            'Tiny: 1 vCPU, 1 GB Memory',
+          ),
+        );
+        await vm.modalEditFlavor();
+
+        await browser.wait(
+          until.textToBePresentInElement(editFlavorView.flavorDropdownText, 'Tiny'),
+        );
+        await selectDropdownOptionById('#vm-flavor-modal-flavor-dropdown', 'Custom-link');
```
Comment (rhrazdil) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.flavor.ts]: Here as well

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.flavor.ts (hunk 3 above)
Comment (mareklibra) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.flavor.ts]: done, moved to `editFlavorView.ts`

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.flavor.ts (hunk 4 above)
Comment (mareklibra) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.flavor.ts]: done

File: frontend/packages/kubevirt-plugin/integration-tests/views/virtualMachine.view.ts [hunk 5]
```diff
@@ -50,10 +50,46 @@ export const vmDetailNamespace = (namespace, vmName) =>
...
+      .$('.co-m-label__key')
+      .getText()
+      .then((text) => text === labelKey),
+  );
+  const valueElem = filteredLabel.first().$('.co-m-label__value');
+  const value = await valueElem.getText();
+  return value;
+};
+
+export const xvmDetailLabel = (labelKey) =>
```
Comment (vojtechszocs) [inline frontend/packages/kubevirt-plugin/integration-tests/views/virtualMachine.view.ts]: Just wondering, why `xvm` and not `vm` name prefix?

File: frontend/packages/kubevirt-plugin/integration-tests/tests/models/editFlavorView.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { $ } from 'protractor';
+
+export const flavorDropdownId = '#vm-flavor-modal-flavor-dropdown';
+
+export const modalTitle = $('.modal-title');
```
Comment (vojtechszocs) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/models/editFlavorView.ts]: As per ES module spec, this file gets evaluated only once, when imported (or required).

I'd suggest moving `$(foo)` statements (which I assume depend on the _current_ DOM context) into functions.

```ts
export const modalTitle = () => $('.modal-title');
```

**Comment by vojtechszocs** [comment]
> for #2817

That one is merged now, `master` is synced with 4.3 changes.

Please note a couple of my review comments, otherwise LGTM.

File: frontend/packages/kubevirt-plugin/integration-tests/views/virtualMachine.view.ts (hunk 5 above)
Comment (mareklibra) [reply frontend/packages/kubevirt-plugin/integration-tests/views/virtualMachine.view.ts]: orphaned left-over, removing

File: frontend/packages/kubevirt-plugin/integration-tests/tests/models/editFlavorView.ts (hunk 6 above)
Comment (mareklibra) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/models/editFlavorView.ts]: done

**Comment by vojtechszocs** [comment]
@mareklibra CI fails on lint errors, please fix :smiley:

**Comment by mareklibra** [comment]
linter issue fixed

**Comment by mareklibra** [comment]
Rebased

**Pull request merged**